src/backend/distributed/test/hide_shards.c

```diff
@@ -0,0 +1,37 @@
+/*-------------------------------------------------------------------------
+ *
+ * hide_shards.c
+ *
+ * This file contains functions to provide helper UDFs for hiding
+ * shards from the applications.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "pgstat.h"
+
+#include "distributed/worker_shard_visibility.h"
+
+
+PG_FUNCTION_INFO_V1(set_backend_type);
+
+/*
+ * set_backend_type is an external API to set the MyBackendType and
+ * re-checks the shard visibility.
+ */
+Datum
+set_backend_type(PG_FUNCTION_ARGS)
+{
+	MyBackendType = PG_GETARG_INT32(0);
```

Comment:
uff, well, ok

Reply:
Maybe I can also EnsureSuperuser() at least.

I tried to find some (a) an API that sets MyBackendType (b) low level UDFs for autovacuum/walsender etc so that we can test on shards.

But, couldn't find :/